### ISSUE TITLE
Proposal: rename `wasi` target OS to `wasip1`

### ISSUE THREAD

**Issue by ianprime0509** [ziglang/zig#19581]
Since [WASI preview 2](https://github.com/WebAssembly/WASI/tree/main/preview2) has been officially launched, and implementations of it are available, the term "WASI" by itself has become ambiguous. Eventually, in the future, when WASI is stabilized, the unqualified term "WASI" should refer to the stabilized version.

The name "wasip1" for preview 1 (which is what Zig currently supports) seems to be the preferred name going forward in the WASI ecosystem, see for example https://github.com/WebAssembly/wasi-libc/pull/478

This proposal doesn't cover how Zig might eventually support preview 2 (which would likely involve several other factors, such as [wasi-libc support](https://github.com/WebAssembly/wasi-libc/issues/447) and integration with the [component model](https://github.com/WebAssembly/component-model) on which preview 2 is based). It just proposes renaming the existing `wasi` target OS to eliminate ambiguity and pave the way for future developments in this area.

**Comment by jacobly0** [comment]
Does #19654 fit your needs?

**Comment by ianprime0509** [comment]
Thanks! That seems reasonable to me. One nit, though, is that I think it would make more sense to have WASI preview 1 be version 0.1.0 in that scheme, as preview 2 is [labeled as 0.2.0 in the WIT IDL definitions](https://github.com/WebAssembly/WASI/tree/main/preview2). (Also, my understanding of the target query parsing code suggests to me that OS versions containing `-`, such as `wasi.0.0.0-preview.1`, would not be parseable, as the arch-OS-ABI triple is split on `-`, but I could be missing something)

**Comment by jacobly0** [comment]
> (Also, my understanding of the target query parsing code suggests to me that OS versions containing -, such as wasi.0.0.0-preview.1, would not be parseable, as the arch-OS-ABI triple is split on -, but I could be missing something)

Yeah I was wondering the same thing, but your suggested versioning solves that neatly.

**Comment by andrewrk** [comment]
Solved in a2b834e8c7152f70d71c71107db40b9182909647.

`-target wasm32-wasi.0.1.0` vs `-target wasm32-wasi.0.2.0`
